packages/ui/app/src/search-context/SearchContextProvider.tsx

```diff
@@ -0,0 +1,31 @@
+import { PropsWithChildren, useCallback, useState } from "react";
+import { SearchContext, SearchContextValue } from "./SearchContext";
+
+export declare namespace SearchContextProvider {
+    export type Props = PropsWithChildren<{
+        // ...
+    }>;
+}
+
+export const SearchContextProvider: React.FC<SearchContextProvider.Props> = ({ children }) => {
+    const [isSearchDialogOpen, setIsSearchDialogOpen] = useState(false);
+
+    const openSearchDialog = useCallback(() => {
+        setIsSearchDialogOpen(true);
+    }, []);
+
+    const closeSearchDialog = useCallback(() => {
+        setIsSearchDialogOpen(false);
+    }, []);
+
+    const contextValue = useCallback(
```

Comment:
super nbd but just fyi there's a nice hook in the codebase already called `useBooleanState`

```ts
const {
  value: isSearchDialogOpen,
  setTrue: openSearchDialog,
  setFalse: closeSearchDialog,
  // also has toggleValue
} = useBooleanState(false);
```